File: src/main/java/com/kaala/java/gcd/Gcd.java
package com.kaala.java.gcd;

/**
 * Created by binodnme on 9/8/2016.
 */
public class Gcd {
    public static long gcd(long a, long b) {
        if( b == 0) return a;
        return gcd(b, a%b);
    }

    public static void main(String[] args) {
        System.out.println(gcd(3918848, 1653264));
    }
}

File: src/main/java/com/kaala/kotlin/gcd/Gcd.kt
package com.kaala.kotlin.gcd

/**
 * @author Binod Shrestha <binodshrestha@lftechnology.com> on 5/27/17.
 */

fun gcd(a: Long, b: Long): Long {
    if (b == 0L) return a else return gcd(b, a % b);
}

fun main(args: Array<String>) {
    println(gcd(3918848, 1653264))
}